packages/board-server/src/server/cors.ts

```diff
@@ -43,17 +44,18 @@ export const corsAll = (req: IncomingMessage, res: ServerResponse) => {
   return false;
 };
 
-export const cors = (req: IncomingMessage, res: ServerResponse) => {
+export const cors = (
+  req: IncomingMessage,
+  res: ServerResponse,
+  allowedOrigins: Set<string>
+) => {
+  const headers = { ...CORS_HEADERS };
   const origin = req.headers.origin || "";
-  const isLocalhost = origin.includes("localhost");
   const host = req.headers.host || "";
   const sameOrigin = origin.includes(host);
-  const headers = structuredClone(CORS_HEADERS);
+
   const isAllowed =
-    isLocalhost ||
-    sameOrigin ||
-    origin.length === 0 ||
-    CONFIG?.allow?.includes(origin);
+    sameOrigin || origin.length === 0 || allowedOrigins.has(origin);
```

Comment:
I wonder if we can remove the `sameOrigin || origin.length === 0` part? Seems like the same origin case should be handled by `allowedOrigins`, and `origin.length` should actually not be allowed? (For another PR maybe, just something I noticed skimming this)

Reply:
Yeah, I was wondering about that myself. I might take a look and follow up on some of these conditions. I'm not sure what triggers them. (For example, when is origin empty?)

I want to do some debugging to see what these values are in practice.